include/tvm/tir/stmt.h

```diff
@@ -1059,6 +1059,20 @@ class MatchBufferRegion : public ObjectRef {
  public:
   TVM_DLL explicit MatchBufferRegion(Buffer buffer, BufferRegion source);
 
+  /*!
+   * \brief Convert target buffer access indices to original one.
+   * \param indices The indices of the target buffer
+   * \return The indices of source buffer.
+   */
+  TVM_DLL Array<PrimExpr> ConvertIndices(const Array<PrimExpr>& indices) const;
```

Comment:
ARCH: Given this is more like a utility to lower the IRNode, it does not bear any semantics meaning here. It would be better to separate that from the IR node to keep it minimum. Instead, include these utilities in buffer lowering pass(lower match buffer)